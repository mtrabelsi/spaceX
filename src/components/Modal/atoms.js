import s from 'styled-components'
import {
    primarytColor,
    primaryFontSize,
    fontFamily,
    secondaryFontSize,
    defaultBoxShadow,
    tabletWidth,
    fontify
  } from './../Global'
  
export const ModalMain = s.div`
  ${fontify()};
    width: calc(100% - 60px);
    max-width: calc(${tabletWidth} - 20px);
    min-height: 300px;
    position: absolute;
    top: 50%;
    background: white;
    padding: 10px;
    border-radius: 5px;
    box-shadow: ${defaultBoxShadow};
    z-index: 1;
`
export const ModalControl = s.div`
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
`
export const ModalTitle = s.h1`
`
export const ModalContent = s.section`
  display: flex;
  justify-content: center;
`
export const ModalBody = s.section`
  display: flex;
  flex-direction: column;
  height: calc(100% - 80px);
  justify-content: space-between;
`