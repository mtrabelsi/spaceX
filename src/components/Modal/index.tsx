import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import {
  ModalMain, ModalControl, ModalTitle, ModalContent, ModalBody,
} from './atoms';
import Button from '../Button';

type ModalPropType = {
    onSubmitHandle: (e: React.MouseEvent) => void,
    onCloseHandle: (e: React.MouseEvent) => void,
    title?: string
}

const Modal : React.FC<RouteComponentProps | ModalPropType> = (props) => {
  const { children } = props;
  const { onSubmitHandle, onCloseHandle, title } = props as ModalPropType;
  return (
    <ModalMain>
      <ModalTitle>{title || 'title not set?'}</ModalTitle>
      <ModalBody>
        <ModalContent>
          {children || 'Empty content'}
        </ModalContent>
        <ModalControl>
          <Button value="Submit" onClick={onSubmitHandle} />
          <Button isDanger value="Close" onClick={onCloseHandle} />
        </ModalControl>
      </ModalBody>
    </ModalMain>
  );
};

export default Modal;
