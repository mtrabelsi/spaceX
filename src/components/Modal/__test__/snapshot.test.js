import React from 'react';
import renderer from 'react-test-renderer';
import Modal from '..';

it('Should consider all props and generate unique html based on that', () => {
  const tree = renderer
    .create(
      <Modal
        title="sample title"
        onCloseHandle={() => console.log('should close')}
        onSubmitHandle={() => console.log('should submit')}
        history={{ goBack: () => {} }}
      >
        <span>some modal child data</span>
      </Modal>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
