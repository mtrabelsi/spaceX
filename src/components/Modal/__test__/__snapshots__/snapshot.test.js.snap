// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should consider all props and generate unique html based on that 1`] = `
<div
  className="sc-AxjAm eSCnPG"
>
  <h1
    className="sc-AxiKw cydMkg"
  >
    sample title
  </h1>
  <section
    className="sc-AxhUy hrDmoG"
  >
    <section
      className="sc-AxhCb fRQPFI"
    >
      <span>
        some modal child data
      </span>
    </section>
    <div
      className="sc-AxirZ cqzgpd"
    >
      <button
        className="sc-AxgMl dNSLJb"
        onClick={[Function]}
      >
        <span
          className="sc-AxheI fqGtiw"
        >
          Submit
        </span>
      </button>
      <button
        className="sc-AxgMl jbpKxJ"
        onClick={[Function]}
      >
        <span
          className="sc-AxheI fqGtiw"
        >
          Close
        </span>
      </button>
    </div>
  </section>
</div>
`;
